### PR TITLE
Add batch in bloomfilter indexer

### PR DESCRIPTION
**Pull request by koseoyoung**
if not use batch, bf indexer rebuilding takes est. 30-40 hours 
if have a batch, bf indexer rebuilding takes 15 hours.

### CODE REVIEW

File: blockindex/bloomfilterindexer.go [hunk 1]
```diff
@@ -206,24 +211,26 @@ func (bfx *bloomfilterIndexer) blockBloomFilter(blockNumber uint64) (bloom.Bloom
...
 func (bfx *bloomfilterIndexer) delete(blockNumber uint64) error {
 	rangeBloomfilterKey := bfx.rangeBloomfilterKey(blockNumber)
-	if err := bfx.kvStore.Delete(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey)); err != nil {
-		return errors.Wrapf(err, "failed to delete RangeBloomFilter")
-	}
-	if err := bfx.kvStore.Delete(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber)); err != nil {
-		return errors.Wrapf(err, "failed to delete BlockBloomFilter")
+	bfx.flusher.KVStoreWithBuffer().MustDelete(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey))
+	bfx.flusher.KVStoreWithBuffer().MustDelete(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber))
+	bfx.flusher.KVStoreWithBuffer().MustDelete(RangeBloomFilterNamespace, []byte(CurrentHeightKey))
```
Comment (CoderZhi) [inline blockindex/bloomfilterindexer.go:216]: use "Delete"

File: blockindex/bloomfilterindexer.go [hunk 2]
```diff
@@ -206,24 +211,26 @@ func (bfx *bloomfilterIndexer) blockBloomFilter(blockNumber uint64) (bloom.Bloom
...
 	rangeBloomfilterKey := bfx.rangeBloomfilterKey(blockNumber)
-	if err := bfx.kvStore.Put(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey), bfx.curRangeBloomfilter.Bytes()); err != nil {
-		return errors.Wrapf(err, "failed to put RangeBloomFilter")
-	}
-	if err := bfx.kvStore.Put(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber), blkBloomfilter.Bytes()); err != nil {
-		return errors.Wrapf(err, "failed to put BlockBloomFilter")
+	bfx.flusher.KVStoreWithBuffer().MustPut(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey), bfx.curRangeBloomfilter.Bytes())
+	bfx.flusher.KVStoreWithBuffer().MustPut(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber), blkBloomfilter.Bytes())
+	bfx.flusher.KVStoreWithBuffer().MustPut(RangeBloomFilterNamespace, []byte(CurrentHeightKey), byteutil.Uint64ToBytes(blockNumber))
+
```
Comment (CoderZhi) [inline blockindex/bloomfilterindexer.go:229]: use "Put"

File: blockindex/bloomfilterindexer.go [hunk 3]
```diff
@@ -85,7 +87,7 @@ func (bfx *bloomfilterIndexer) Start(ctx context.Context) error {
 			}
 		}
 	case db.ErrNotExist:
-		if err = bfx.kvStore.Put(RangeBloomFilterNamespace, []byte(CurrentHeightKey), byteutil.Uint64ToBytes(0)); err != nil {
+		if err = bfx.flusher.KVStoreWithBuffer().Put(RangeBloomFilterNamespace, []byte(CurrentHeightKey), byteutil.Uint64ToBytes(0)); err != nil {
```
Comment (CoderZhi) [inline blockindex/bloomfilterindexer.go:90]: flush?

File: blockindex/bloomfilterindexer.go [hunk 4]
```diff
@@ -209,27 +211,25 @@ func (bfx *bloomfilterIndexer) blockBloomFilter(blockNumber uint64) (bloom.Bloom
 
 func (bfx *bloomfilterIndexer) delete(blockNumber uint64) error {
 	rangeBloomfilterKey := bfx.rangeBloomfilterKey(blockNumber)
-	bfx.batch.Delete(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey), "failed to delete RangeBloomFilter at block number %d", blockNumber)
-	bfx.batch.Delete(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber), "failed to delete BlockBloomFilter at block number %d", blockNumber)
-	bfx.batch.Put(RangeBloomFilterNamespace, []byte(CurrentHeightKey), byteutil.Uint64ToBytes(rangeBloomfilterKey-bfx.rangeSize), "failed to update current height")
+	bfx.flusher.KVStoreWithBuffer().MustDelete(RangeBloomFilterNamespace, byteutil.Uint64ToBytes(rangeBloomfilterKey))
+	bfx.flusher.KVStoreWithBuffer().MustDelete(BlockBloomFilterNamespace, byteutil.Uint64ToBytes(blockNumber))
+	bfx.flusher.KVStoreWithBuffer().MustDelete(RangeBloomFilterNamespace, []byte(CurrentHeightKey))
```
Comment (CoderZhi) [inline blockindex/bloomfilterindexer.go:216]: Put?

Comment (koseoyoung) [reply blockindex/bloomfilterindexer.go:216]: thanks for catching out!

File: blockindex/bloomfilterindexer.go (hunk 1 above)
Comment (koseoyoung) [reply blockindex/bloomfilterindexer.go:216]: anyway "Delete" always return nil and behaves the same as "MustDelete" - why do "Delete" is better?

Comment (koseoyoung) [reply blockindex/bloomfilterindexer.go]: addressed on the latest commit!

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**